autogpt_platform/frontend/src/components/cron-scheduler.tsx

```diff
@@ -0,0 +1,460 @@
+import React, { useEffect, useState } from "react";
+import {
+  Select,
+  SelectContent,
+  SelectItem,
+  SelectTrigger,
+  SelectValue,
+} from "@/components/ui/select";
+import { Label } from "@/components/ui/label";
+import { Input } from "@/components/ui/input";
+import { Button } from "@/components/ui/button";
+import { useToast } from "@/components/ui/use-toast";
+import { Separator } from "@/components/ui/separator";
+import { makeCronExpression } from "@/lib/cron-expression-utils";
+import { Dialog, DialogContent, DialogTitle } from "@/components/ui/dialog";
+
+type CronFrequency =
+  | "minute"
+  | "hour"
+  | "daily"
+  | "weekly"
+  | "monthly"
+  | "yearly"
+  | "custom";
+
+export function CronScheduler({
+  onCronExpressionChange,
+}: {
+  onCronExpressionChange: (cronExpression: string) => void;
+}): React.ReactElement {
+  const [frequency, setFrequency] = useState<CronFrequency>("daily");
+  const [selectedDays, setSelectedDays] = useState<number[]>([]);
+  const [selectedTime, setSelectedTime] = useState<string>("09:00");
+  const [showCustomDays, setShowCustomDays] = useState<boolean>(false);
+  const [selectedMinute, setSelectedMinute] = useState<string>("0");
+  const [customInterval, setCustomInterval] = useState<{
+    value: number;
+    unit: "minutes" | "hours" | "days";
+  }>({ value: 1, unit: "minutes" });
```

Comment:
Most likely all this can be a single state?
```tsx
const [cronState, setCronState] = useState({
  frequency: "daily" as CronFrequency,
  selectedDays: [] as number[],
  selectedTime: "09:00",
  selectedMinute: "0",
  customInterval: { value: 1, unit: "minutes" as const },
});
```
Then if you need to update a single value:
```tsx
setCronState((prev) => ({ ...prev, frequency: "weekly" }));
```
In this way the component state is easier to read/understand. In general is a good practice to group the state in an object when you start to have more than 3 `useState` per component 👍🏽